src/docfx/lib/ProcessUtility.cs

```diff
@@ -98,5 +107,56 @@ public static Task<string> Execute(string fileName, string commandLineArgs, stri
 
             return tcs.Task;
         }
+
+        /// <summary>
+        /// Provide a process lock function based on locking file
+        /// </summary>
+        /// <param name="action">The action you want to lock</param>
+        /// <param name="lockPath">The lock file path, default is a file with GUID name</param>
+        /// <param name="retry">The retry count, default is 600 times</param>
+        /// <param name="retryTimeSpanInterval">The retry interval, default is 1 seconds</param>
+        /// <returns>The task status</returns>
+        public static async Task ProcessLock(Func<Task> action, string lockPath, int retry = 600, TimeSpan? retryTimeSpanInterval = null)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(lockPath));
+            Debug.Assert(!PathUtility.FilePathHasInvalidChars(lockPath));
+            Directory.CreateDirectory(Path.GetDirectoryName(lockPath));
```

Comment:
>CreateDirectory [](start = 22, length = 15)

it throws error when directory name is empty

Reply:
thanks for the reminder. 

---
In reply to: [182645197](https://github.com/dotnet/docfx/pull/2651#discussion_r182645197) [](ancestors = 182645197)